src/core/internal/lifetime.d

```diff
@@ -88,30 +88,33 @@ if (is(UT == core.internal.traits.Unqual!UT))
 Emplaces T.init.
 In contrast to `emplaceRef(chunk)`, there are no checks for disabled default
 constructors etc.
+The keepContext template argument instructs this function to keep a nested struct's
+context pointer intact (used by move et al.).
 +/
-void emplaceInitializer(T)(scope ref T chunk) nothrow pure @trusted
+void emplaceInitializer(T,bool keepContext = false)(scope ref T chunk) nothrow pure @trusted
     if (!is(T == const) && !is(T == immutable) && !is(T == inout))
 {
     import core.internal.traits : hasElaborateAssign;
 
-    static if (!hasElaborateAssign!T && __traits(compiles, chunk = T.init))
+    static if (keepContext) {
+        static assert(is(T == struct) && __traits(isNested, T),
+                      "keepContext should only be set to true when keeping a nested struct's context pointer is required");
+        enum sizeToBlit = T.sizeof - (void*).sizeof;
```

Comment:
This wasn't and isn't fully correct and should probably be `enum sizeToBlit = T.tupleof[$-1].offsetof`:
```D
void foo()
{
    int x;
	struct S
	{
    	align(64) void* data;
        void bar() { x = 123; }
	}
    static assert(S.sizeof == 64);
    static assert(S.tupleof[$-1].offsetof == size_t.sizeof);
}
```